From aeac0cb7b3c99ebfceca322501dbbbdbc5f0a572 Mon Sep 17 00:00:00 2001
From: Han Gao <gaohan@iscas.ac.cn>
Date: Sun, 10 May 2026 23:43:16 +0800
Subject: [PATCH 095/102] RUYI: riscv: dts: pine64-starpro64: Add Reserve
 Memory Node

Reserve a 512 MiB CMA region as the default DMA pool on the Pine64
StarPro64 board.

Signed-off-by: Han Gao <gaohan@iscas.ac.cn>
---
 .../boot/dts/eswin/eic7700-pine64-starpro64.dts  | 16 ++++++++++++++++
 1 file changed, 16 insertions(+)

diff --git a/arch/riscv/boot/dts/eswin/eic7700-pine64-starpro64.dts b/arch/riscv/boot/dts/eswin/eic7700-pine64-starpro64.dts
index b7ff07efa6bf..44d71a15b729 100644
--- a/arch/riscv/boot/dts/eswin/eic7700-pine64-starpro64.dts
+++ b/arch/riscv/boot/dts/eswin/eic7700-pine64-starpro64.dts
@@ -5,6 +5,8 @@
 
 /dts-v1/;
 
+#define CMA_SIZE        0x20000000
+
 #include "eic7700.dtsi"
 
 / {
@@ -23,6 +25,20 @@ chosen {
 		stdout-path = "serial0:115200n8";
 	};
 
+	reserved-memory {
+		#address-cells = <2>;
+		#size-cells = <2>;
+		ranges;
+
+		linux,cma {
+			compatible = "shared-dma-pool";
+			reusable;
+			size = <0x0 CMA_SIZE>;
+			alignment = <0x0 0x1000>;
+			linux,cma-default;
+		};
+	};
+
 	vcc_1v8: vcc1v8 {
 		compatible = "regulator-fixed";
 		regulator-name = "vcc1v8";
-- 
2.47.3

